### DISCUSSION TITLE
Raise Compile Error When Using DateTime.Now

### DISCUSSION THREAD

**Discussion by Alirexaa** [dotnet/roslyn#74042] [Q&A] [answered]
Hi, is there any way to raise a compile error when Using `DateTime.Now`?
and force to use `DateTime.UtcNow` instead of `DateTime.Now`?

**Comment by CyrusNajmabadi** [comment, marked as answer]
Yes. You can write an analyzer that blocks this. Or look at the banned-api analyzer in GitHub.com/dotnet/roslyn-analyzers

**Comment by Alirexaa** [comment]
@CyrusNajmabadi thanks for your answer. The banned-api analyzer works, and I am interested in writing an analyzer. is there any article or manual about it?

**Reply by WeihanLi** [reply]
https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix

Hope it helps
